### DISCUSSION TITLE
how to convert bin to uf2 for itsy bitsy?

### DISCUSSION THREAD

**Discussion by kimgalibert** [shanteacontrols/OpenDeck#277] [Q&A]
i have found two tools to convert bin files to uf2 for installation on a adafruit itsy bitsy, neither works on linux or windows.

**Comment by paradajz** [comment]
Which ItsyBitsy?

**Reply by kimgalibert** [reply]
it is an adafruit itsy bitsy nrf52840.

**Reply by paradajz** [reply]
I don't understand what do you need uf2 file for? You need to flash that board with an external programmer. Converting the OpenDeck binary to uf2 won't work.

**Reply by kimgalibert** [reply]
please elaborate on the procedure, there is nothing on the site re: flashing the itsy bitsy nrf52840.

i have it attached to a programming board.  i have the bin file.  

i've tried nrfconnect.

**Comment by paradajz** [comment]
See https://github.com/shanteacontrols/OpenDeck/discussions/168#discussioncomment-3447084

**Reply by kimgalibert** [reply]
please confirm that i will need a second adafruit itsy bitsy nrf52840 in order to program the first.  

also that no other hardware/software must be purchased in order to proceed.

**Reply by Bosswhite12** [reply]
Hello Please can you tell me if you ever worked out how to program the itsy bitsy?
Thanks Nick

**Reply by kimgalibert** [reply]
i haven't tried again.

i don't mind buying what's needed.  and i don't mind terribly if a board gets wasted.

but the time involved is the killer.  waiting for parts.  and then waiting again once you find out you need yet another device.  

i'd contribute $ to fund the writing of an itsy bitsy guide.  there is no link to donate.  maybe the best way to support shantea is to buy one of their boards?

**Reply by basst22778** [reply]
did you read the adafruit manual about reflashing the itsy bitsy? https://learn.adafruit.com/how-to-program-samd-bootloaders

**Comment by paradajz** [comment]
The instructions are now available here:
https://github.com/shanteacontrols/OpenDeck/wiki/Flashing-nRF-boards

**Reply by Bosswhite12** [reply]
Many thanks for this. Nick
